src/lib_io.c

```diff
@@ -164,9 +164,14 @@ static void io_file_readall(lua_State *L, FILE *fp)
   MSize m, n;
   for (m = LUAL_BUFFERSIZE, n = 0; ; m += m) {
     char *buf = lj_buf_tmp(L, m);
-    n += (MSize)fread(buf+n, 1, m-n, fp);
-    if (n != m) {
-      setstrV(L, L->top++, lj_str_new(L, buf, (size_t)n));
+    for (;;) {
+      n += (MSize)fread(buf+n, 1, m-n, fp);
+      if (n == m) break;
+      else if (ferror(fp)) {
+        if (errno == EINTR) { clearerr(fp); continue; }
```

Comment:
I think the previous behavior is to read until the end of the stream or until an error occurred. here we throw away all the already read data when an error happens. This looks wrong to me.

Reply:
I don’t see any path when an error occurs where this value would be used. From my understanding, the previous behavior simply did not consider errors, and thus did not need to deal with such cases.

Reply:
io_file_readall() is a helper function of io_file_read() which check if error take place.

Interruption should not be considered as a real error, also it's difficult to resume read/write in io_file_read(), so I guess this piece of code needs some change.

That being said, I don't think this fix is enough. For one thing, other helper function (say io_file_readnum()) have similar problem, and this patch does not cover them. On the other hand, this is the problem specific the OR environment. Maybe you can explain the problem to Luajit mailing list and solicit comment from broader audiences?

Reply:
Yes, this fix is only contained to readall. I know there are other areas that require similar attention.